subsys/net/lib/mqtt_socket/mqtt_internal.h

```diff
@@ -0,0 +1,439 @@
+ *        keep the connection alive are governed by this value.
+ */
+#define MQTT_KEEPALIVE CONFIG_MQTT_KEEPALIVE
+
+/**@brief Maximum MQTT packet size that can be sent (including the fixed and
+ *        variable header).
+ */
+#define MQTT_MAX_PACKET_LENGTH CONFIG_MQTT_MAX_PACKET_LENGTH
+
+/**@brief Fixed header minimum size. Remaining length size is 1 in this case. */
+#define MQTT_FIXED_HEADER_SIZE 2
+
+/**@brief Maximum size of the fixed header. Remaining length size is 4 in this
+ *        case.
+ */
+#define MQTT_FIXED_HEADER_EXTENDED_SIZE 5
+
+/**@brief MQTT Control Packet Types. */
+#define MQTT_PKT_TYPE_CONNECT     0x10
+#define MQTT_PKT_TYPE_CONNACK     0x20
+#define MQTT_PKT_TYPE_PUBLISH     0x30
+#define MQTT_PKT_TYPE_PUBACK      0x40
+#define MQTT_PKT_TYPE_PUBREC      0x50
+#define MQTT_PKT_TYPE_PUBREL      0x60
+#define MQTT_PKT_TYPE_PUBCOMP     0x70
+#define MQTT_PKT_TYPE_SUBSCRIBE   0x82 /* QoS 1 for subscribe */
+#define MQTT_PKT_TYPE_SUBACK      0x90
+#define MQTT_PKT_TYPE_UNSUBSCRIBE 0xA2
+#define MQTT_PKT_TYPE_UNSUBACK    0xB0
+#define MQTT_PKT_TYPE_PINGREQ     0xC0
+#define MQTT_PKT_TYPE_PINGRSP     0xD0
+#define MQTT_PKT_TYPE_DISCONNECT  0xE0
+
+/**@brief Masks for MQTT header flags. */
+#define MQTT_HEADER_DUP_MASK     0x08
+#define MQTT_HEADER_QOS_MASK     0x06
+#define MQTT_HEADER_RETAIN_MASK  0x01
+
+
+/**@brief Masks for MQTT header flags. */
+#define MQTT_CONNECT_FLAG_CLEAN_SESSION   0x02
+#define MQTT_CONNECT_FLAG_WILL_TOPIC      0x04
+#define MQTT_CONNECT_FLAG_WILL_RETAIN     0x20
+#define MQTT_CONNECT_FLAG_PASSWORD        0x40
+#define MQTT_CONNECT_FLAG_USERNAME        0x80
+
+#define MQTT_CONNACK_FLAG_SESSION_PRESENT 0x01
+
+/**@brief Size of mandatory header of MQTT packet. */
+#define MQTT_PKT_HEADER_SIZE 2
+
+/**@brief Maximum payload size of MQTT packet. */
+#define MQTT_MAX_PAYLOAD_SIZE 0x0FFFFFFF
+
+/**@brief Maximum size of variable and payload in the packet. */
+#define MQTT_MAX_VARIABLE_HEADER_N_PAYLOAD \
+	(MQTT_MAX_PACKET_LENGTH - MQTT_FIXED_HEADER_EXTENDED_SIZE)
+
+/**@brief Defines size of uint8 in bytes. */
+#define SIZE_OF_UINT8 1
+
+/**@brief Defines size of uint16 in bytes. */
+#define SIZE_OF_UINT16 2
+
+/**@brief Computes total size needed to pack a UTF8 string. */
+#define GET_UT8STR_BUFFER_SIZE(STR) (SIZE_OF_UINT16 + (STR)->size)
+
+/**@brief Computes total size needed to pack a binary stream. */
+#define GET_BINSTR_BUFFER_SIZE(STR) ((STR)->len)
+
+/**@brief Unsubscribe packet size. */
+#define MQTT_UNSUBSCRIBE_PKT_SIZE 4
+
+/**@brief Sets MQTT Client's state with one indicated in 'STATE'. */
+#define MQTT_SET_STATE(CLIENT, STATE) ((CLIENT)->state |= (STATE))
+
+/**@brief Sets MQTT Client's state exclusive to 'STATE'. */
+#define MQTT_SET_STATE_EXCLUSIVE(CLIENT, STATE) ((CLIENT)->state = (STATE))
+
+/**@brief Verifies if MQTT Client's state is set with one indicated in 'STATE'.
+ */
+#define MQTT_VERIFY_STATE(CLIENT, STATE) ((CLIENT)->state & (STATE))
```

Comment:
I don't anything here which "verifies" the state. Given existing SET/RESET vocab around, MQTT_IS_STATE or MQTT_HAS_STATE would sound good.